File: AtCorder2/A05.java
import java.util.*;

//A05
public class Main {
  public static void main(String[] args) {
    Scanner scanner = new Scanner(System.in);
    int n = scanner.nextInt();
    int k = scanner.nextInt();
    scanner.close();
    long answer = 0L;
    int diff;
    for (int i = 1; i <= n; i++) {
      for (int j = 1; j <= n; j++) {
        diff = k - i - j;
        if (diff >= 1 && diff <= n) {
          answer++;
        }
      }
    }
    System.out.println(answer);
  }
}

File: AtCorder2/A05.py
# 鉄則問題集A05（）
import sys

# sys.setrecursionlimit(10**7)

def solve():
    n, k = map(int, input().split())
    answer = 0
    for i in range(1,n + 1):
        for j in range(1,n + 1):
            diff = k - i - j
            if 1 <= diff <= n:
                answer += 1
    print(answer)

if __name__ == "__main__":
    solve()


